labcontrol/gui/js_tests/addPlateModal_tests.js

```diff
@@ -5,150 +5,162 @@
-
-        // before running function under test, qunit-fixture has no div
-        // with explicit modal id and no table with explicit table name
-        assert.equal($(modalIdDivSelectorStr).length, 0);
-        assert.equal($(tableIdSelectorStr).length, 0);
-
-        insertPlateModalDiv(parentId, modalId, tableId);
-
-        // after running function under test, table has these elements
-        assert.equal($(modalIdDivSelectorStr).length, 1);
-        assert.equal($(tableIdSelectorStr).length, 1);
-    });
+QUnit.module("insertPlateModalDiv", function(hooks) {
+  // Verify correct behavior using default arguments
+  QUnit.test("default arguments", function(assert) {
+    //jquery selector strings
+    let modalIdDivSelectorStr = "body div#addPlateModal";
+    let tableIdSelectorStr = "body table#searchPlateTable";
+
+    // before running function under test, body has no div with
+    // default modal id and no table with default table name
+    assert.equal($(modalIdDivSelectorStr).length, 0);
+    assert.equal($(tableIdSelectorStr).length, 0);
+
+    insertPlateModalDiv();
+
+    // after running function under test, table has these elements
+    assert.equal($(modalIdDivSelectorStr).length, 1);
+    assert.equal($(tableIdSelectorStr).length, 1);
+
+    // NOTE: unlike python unit tests, qunit does not abort the whole
+    // test case when an assert fails.  Therefore, the cleanup can
+    // simply be done in-line at the end of the test.
+    // Also note that this clean-up can't be done automatically as the dom
+    // objects were added (intentionally) to body rather than to the
+    // qunit-fixture div.
+    $(modalIdDivSelectorStr).remove();
+  });
+
+  // Verify correct behavior when arguments are explicitly specified
+  QUnit.test("explicit arguments", function(assert) {
+    //explicit names
+    let modalId = "myModal";
+    let tableId = "myPlateTable";
+    let parentId = "qunit-fixture";
+
+    //jquery selector strings
+    let modalIdDivSelectorStr = "#" + parentId + " div#" + modalId;
+    let tableIdSelectorStr = "#" + parentId + " table#" + tableId;
+
+    // before running function under test, qunit-fixture has no div
+    // with explicit modal id and no table with explicit table name
+    assert.equal($(modalIdDivSelectorStr).length, 0);
+    assert.equal($(tableIdSelectorStr).length, 0);
+
+    insertPlateModalDiv(parentId, modalId, tableId);
+
+    // after running function under test, table has these elements
+    assert.equal($(modalIdDivSelectorStr).length, 1);
+    assert.equal($(tableIdSelectorStr).length, 1);
+  });
 });
 
 // Tests of functions that depend on the plate modal div having already
 // been inserted
-QUnit.module("modal-div-dependent tests", function (hooks) {
-    hooks.beforeEach(function (assert) {
-        insertPlateModalDiv("qunit-fixture");
+QUnit.module("modal-div-dependent tests", function(hooks) {
+  hooks.beforeEach(function(assert) {
+    insertPlateModalDiv("qunit-fixture");
+  });
```

Comment:
@fedarko Just to confirm, there is supposed to be a closing brace here, before ');'? I'm asking because this has the effect of taking everything that was in beforeEach() below, and moving it out.

Reply:
Note for future reference: @charles-cowart and I looked at the history of this file and verified that these changes were ok.